torch/testing/_internal/common_distributed.py

```diff
@@ -353,6 +353,22 @@ def skip_if_win32():
     )
 
 
+def sm_is_or_higher_than(device: torch.device, major: int, minor: int) -> bool:
+    """
+    Returns True if the device's compute capability is (major, minor) or higher.
+    Error out if the device is not a CUDA device.
+    Returns False if device is a RoCM device.
+    """
+    if device.type != "cuda":
+        raise ValueError("sm_is_or_later() is only supported for CUDA devices")
+
+    if torch.version.hip is not None:
+        # ROCm devices may have different compute capability codes
+        return False
+
+    return torch.cuda.get_device_capability(device) >= (major, minor)
```

Comment:
maybe we can reuse `SM80OrLater` from `torch/testing/_internal/common_cuda.py`? would be nice to concentrate these check logic in one place haha

Reply:
also `SM80OrLater` seems to not need to take `device` as input, maybe useful for moving the "skip-test" check to become a decorator on the unit test as well

Reply:
Yeah, I have seen those and thought about using them earlier.
But the short answer is no.
If we use them in any skip deco, they will create a device context on GPU 0.
because in the skip deco, we don’t know the device yet, hence SM80OrLater would pick device 0.
That’s why I had my own version which takes “device” as an argument, and only use “in test”.

Reply:
got it, it makes sense, thanks!